### DISCUSSION TITLE
Recursive struct inits

### DISCUSSION THREAD

**Discussion by medvednikov** [vlang/v#11803] [Ideas]
This used to work in V1, and is very useful, especially now that reference fields are mandatory to init:

```
app := &App{
  ctx: &Context{
    app: app
  }
}
```

this should generate something like

```
mut app := &App{}
app.ctx = &Context{ app: app }
```

Right now it results in 

`error: checker: too many expr levels: 201`

**Comment by ntrel** [comment]
> app := &App{
  ctx: &Context{
    app: app

This breaks the expected evaluation order and would be a special case just for reference fields. If you change expectations of assignment evaluation order, people will expect other things to work, e.g. recursive anonymous functions:
```v
f := fn(i int){
  println(i)
  if i > 0 {f(i-1)} // use `f` identifier before `f` is assigned
}
```
The above could be useful, but do we want to support it in all cases? In general I think it could cause problems. A complex initializer may accidentally reference the assignee from copying and pasting, and V should detect that as an error.
